### ISSUE TITLE
$view_topic_url capture the wrong url with some oficial extensions

### ISSUE THREAD

**Issue by Leinad4Mind** [rmcgirr83/phpBB-3.1-topfive#16]
I have setup the top5 ext to show the 5 most recent posts on the forum and it works perfectly, the only issue is i am also using the ext 'pages' to host a blank page with my content on it while still containing all of the structure of the forum layout. Heres the problem, the Top 5 links try and find the post with an extra directory in the url.

Heres the top5 link on a normal forum page working:
http:/myforum.com/viewtopic.php?f=2&p=525#p525

and here is the links the top5 takes you while in the 'pages' additional page:
http:/myforum.com/page/viewtopic.php?f=2&p=525#p525

Because the custom pages appear to reside in the 'page' directory the top5 is now looking at the wrong urls, how can i make the top5 links correct JUST when its used on the custom pages?

Basically, the "$view_topic_url" appears with the /page/, and it shouldn't.

Best Regards

**Comment by rmcgirr83** [comment]
Not really sure what you mean.  What does the top five extension have to do with the pages extension?  How are you incorporating the two?  What is the real link to the forum so I can see what it is you are trying to accomplish?

**Comment by Leinad4Mind** [comment]
It's very simple. I'll show you on this test forum:

http://www.leinad4mind.top/ - In this link the top5 works as it should.

http://www.leinad4mind.top/page/test - on this page (using Pages extension) on the "recent topics" if I click on any link, it goes to, for example: http://www.leinad4mind.top/page/viewtopic.php?f=2&p=52#p52

As we can see it appears http://www.leinad4mind.top/page/ instead of just http://www.leinad4mind.top

Best Regards

**Comment by rmcgirr83** [comment]
But how are you including the top five into the pages extension?

**Comment by Leinad4Mind** [comment]
I'm just including on the footer of phpbb.

**Comment by rmcgirr83** [comment]
It is because `$this->phpbb_root_path` is being overwritten somehow by the pages extension.  Are the names of the users clickable and don't also include `\page\`?

**Comment by Leinad4Mind** [comment]
The usernames are clickable but doesn't include /page/

**Comment by rmcgirr83** [comment]
Well that's really strange.

**Comment by Leinad4Mind** [comment]
Very xD

The phpbb_root_path shouln't have the /page/

I could just see that it's grabbing the "$phpbb_root_path" but I don't know where this value comes from. The construct is passing the value, but from where, do you know?

**Comment by rmcgirr83** [comment]
from the services file `%core.root_path%`

**Comment by Leinad4Mind** [comment]
I've found the "bug"

this lines:
`			$view_topic_url = append_sid("{$this->phpbb_root_path}viewtopic.$this->php_ext", 'f=' . $row['forum_id'] . '&amp;p=' . $row['topic_last_post_id'] . '#p' . $row['topic_last_post_id']);
			$forum_name_url = append_sid("{$this->phpbb_root_path}viewforum.$this->php_ext", 'f=' . $row['forum_id']);`

have this variable: $this->phpbb_root_path

but on the top of the file (topfive.php) you have: $this->root_path = $phpbb_root_path;
So it should be $this->root_path and not $this->phpbb_root_path.

After fixing this, it works :)

**Comment by rmcgirr83** [comment]
Nice find.  Surprised that it wasn't picked up during validation.  Time for a new version.
